content/yaml-publishing/cloudflare-pages.md

```diff
@@ -0,0 +1,108 @@
+
+See the offical [docs](https://developers.cloudflare.com/workers/wrangler/ci-cd/).
+
+### Configure environment variables
+
+1. Open your Codemagic app settings, and go to the **Environment variables** tab.
+2. Enter the desired **_Variable name_**, e.g. `CLOUDFLARE_API_TOKEN`.
+3. Enter the desired variable value as **_Variable value_**.
+4. Enter the variable group name, e.g. **_cloudflare_credentials_**. Click the button to create the group.
+5. Make sure the **Secure** option is selected.
+6. Click the **Add** button to add the variable.
+7. Repeat the steps to add the `CLOUDFLARE_ACCOUNT_ID`.
+
+8. Add the variable group to your `codemagic.yaml` file
+{{< highlight yaml "style=paraiso-dark">}}
+  environment:
+    groups:
+      - cloudflare_credentials # <-- (Includes CLOUDFLARE_API_TOKEN, CLOUDFLARE_ACCOUNT_ID)
+{{< /highlight >}}
+
+
+## Publish to Cloudflare Pages
+
+After you have created your page from the [Cloudflare dashboard](https://dash.cloudflare.com/) and gave it a name, you can configure automatic publishing in your `codemagic.yaml`.
+
+First we need to install `wingler`, which is a command-line tool for building with Cloudflare developer products, and then publish our website.
+
+
+Add the following script to your `publishing` section:
+{{< highlight yaml "style=paraiso-dark">}}
+publishing:
+    scripts:
+    - name: Install wrangler
+        script: npm install -g wrangler    
+    - name: Deploy to Cloudflare Pages
+        script: | 
+        wrangler pages publish <path/to/your/build/folder/> --project-name <your-project-name> --branch <branch-name>
+{{< /highlight >}}
+
+{{<notebox>}}
+You can change the default commit message which is your current git commit message by using `--commit-message "Your commit message"`
+{{</notebox>}}
+
+If the deployment is complete then you should be able to browse your website using the URL at the last line of the log.
+
+## Flutter web sample workflow
+Here's a workflow for building a Flutter web application and then publish it to Cloudflare Pages.
+{{< highlight yaml "style=paraiso-dark">}}
+  web-workflow:
+    name: Web Workflow
+    instance_type: linux_x2
+    environment:
+      groups:
+        - cloudflare # <-- (Includes CLOUDFLARE_API_TOKEN, CLOUDFLARE_ACCOUNT_ID)
+      flutter: stable
+    scripts:
+      - name: Get Flutter packages
+        script: flutter packages pub get
+      - flutter config --enable-web
+      - name: Build Web
+        script: | 
+          flutter build web --release
+      - name: Gather the web files
+        script: | 
+          cd build/web
+          7z a -r ../web.zip ./*
+    artifacts:
+      - build/web.zip
+    publishing:
+      scripts:
+        - name: Install wrangler
+          script: npm install -g wrangler
+        - name: Deploy to Cloudflare Pages
+          script: | 
+            wrangler pages publish build/web/ --project-name my-flutter-pages --branch production
+      slack:
+        channel: "#test-codemagic1"
```

Comment:
I'd probably use a more generic name like 'builds'.

Reply:
Do you mean the branch name or what? @nevercode-kevin

Reply:
I mean for the Slack channel name.